x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -261,16 +261,19 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		}
 
 		details["relayNumber"] = strconv.FormatUint(relay.RelayNum, 10)
-		// differentiate between different relays by providing the index in the keys
-		successDetails := appendRelayPaymentDetailsToEvent(details, uint64(relayIdx))
-		// calling the same event repeatedly within a transaction just appends the new keys to the event
-		utils.LogLavaEvent(ctx, logger, types.RelayPaymentEventName, successDetails, "New Proof Of Work Was Accepted")
 
-		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay)
+		cuAfterQos := uint64(reward.Quo(coinsPerCu).TruncateInt64())
+		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay, cuAfterQos)
```

Comment:
the name of the function is misleading since it's also crediting CU to the provider not only charging

Reply:
done. see https://github.com/lavanet/lava/commit/1916d2933243bfea20dc6e1f70af7ca822a907b2